b2/bucket.py

```diff
@@ -271,6 +275,97 @@ def upload_local_file(
         upload_source = UploadSourceLocalFile(local_path=local_file, content_sha1=sha1_sum)
         return self.upload(upload_source, file_name, content_type, file_infos, progress_listener)
 
+    def upload_stream(
+        self,
+        file_name,
+        part_size,
+        content_type=None,
+        file_infos=None,
+    ):
+        """
+        Uploads a file from stdin stream to a B2 file.
+        """
+        #determine if it requires large file upload, > part_size >= minimum_part_size
+        #TODO make part_size user configurable 100MB to 5GB so max file size of 10TB can be reached with 1 to 10000 parts
+        chunk_size = 16*1024 #16KiB chunk size for memory considerations
+        validate_b2_file_name(file_name)
+        file_info = file_infos or {}
+        content_type = content_type or self.DEFAULT_CONTENT_TYPE
+        progress_listener = DoNothingProgressListener()
+
+        temp_fd, temp_path = tempfile.mkstemp()
```

Comment:
if b2 CLI (or a program using the b2 API library) dies for some reason during the upload, the file will leak and consume space in the user system until someone comes and removes the file manually. Using `tempfile.TemporaryFile` would prevent the leak on some systems.